src/core/server/integration_tests/http/router.test.ts

```diff
@@ -670,55 +671,118 @@ describe('Handler', () => {
     `);
   });
 
-  it('returns 400 Bad request if request validation failed', async () => {
-    const { server: innerServer, createRouter } = await server.setup(setupDeps);
-    const router = createRouter('/');
+  describe('returns 400 Bad request if request validation failed', () => {
+    it('@kbn/config-schema', async () => {
+      const { server: innerServer, createRouter } = await server.setup(setupDeps);
+      const router = createRouter('/');
 
-    router.get(
-      {
-        path: '/',
-        validate: {
-          query: schema.object({
-            page: schema.number(),
-          }),
+      router.get(
+        {
+          path: '/',
+          validate: {
+            query: schema.object({
+              page: schema.number(),
+            }),
+          },
         },
-      },
-      (context, req, res) => res.noContent()
-    );
-    await server.start();
+        (context, req, res) => res.noContent()
+      );
+      await server.start();
 
-    const result = await supertest(innerServer.listener)
-      .get('/')
-      .query({ page: 'one' })
-      .expect(400);
+      const result = await supertest(innerServer.listener)
+        .get('/')
+        .query({ page: 'one' })
+        .expect(400);
 
-    expect(result.body).toEqual({
-      error: 'Bad Request',
-      message: '[request query.page]: expected value of type [number] but got [string]',
-      statusCode: 400,
-    });
+      expect(result.body).toEqual({
+        error: 'Bad Request',
+        message: '[request query.page]: expected value of type [number] but got [string]',
+        statusCode: 400,
+      });
 
-    expect(loggingSystemMock.collect(logger).error).toMatchInlineSnapshot(`
-      Array [
+      expect(loggingSystemMock.collect(logger).error).toMatchInlineSnapshot(`
+              Array [
+                Array [
+                  "400 Bad Request",
+                  Object {
+                    "error": Object {
+                      "message": "[request query.page]: expected value of type [number] but got [string]",
+                    },
+                    "http": Object {
+                      "request": Object {
+                        "method": "get",
+                        "path": "/",
+                      },
+                      "response": Object {
+                        "status_code": 400,
+                      },
+                    },
+                  },
+                ],
+              ]
+          `);
+    });
+
+    it('@kbn/zod', async () => {
```

Comment:
Just a basic test that zod validation schemas are functional end-to-end for requests. Some work is required to get "better" error messages from Zod -- see the stringified array in `message` vs the string in `message` above. In my PoC from a while back I did it like [this](https://github.com/elastic/kibana/pull/156357/files#diff-123f8fac4dd9d6418d9b27e42267af57851aaf671ef8cc320ac4ca14d813bf5dR30-R41).

We can consider doing something like that using [Zod error maps](https://zod.dev/ERROR_HANDLING?id=customizing-errors-with-zoderrormap) to get errors that look more like config-schema.

Seems like a follow up item, WDYT @maryam-saeidi ?